File: 2015.2.py
# AoC 2015 Day 2 #

with open("2015.2.in", "r") as file:
    data = file.read()

lines = data.splitlines()

W = [int(line.split("x")[0]) for line in lines]
H = [int(line.split("x")[1]) for line in lines]
L = [int(line.split("x")[2]) for line in lines]

A = [l * w for l, w in zip(L, W)]
B = [w * h for w, h in zip(W, H)]
C = [h * l for h, l in zip(H, L)]

p1 = sum(2*a + 2*b + 2*c + min(a, b, c) for a, b, c in zip(A, B, C))

A = [l + w for l, w in zip(L, W)]
B = [w + h for w, h in zip(W, H)]
C = [h + l for h, l in zip(H, L)]

p2 = sum(min(2 * a, 2 * b, 2 * c) + w*h*l for w, h, l, a, b, c in zip(W, H, L, A, B, C))

print(f"A: {p1}")
print(f"B: {p2}")

File: 2015.2.lua
-- AoC 2015 Day 2 --

local input = assert(io.open"2015.2.in"):read"*a"

local p1, p2 = 0, 0
for line in input:gmatch"[^\n]+" do
	local w, h, l = line:match"(%d*)x(%d+)x(%d+)"
	local a, b, c = 0, 0, 0
	w = tonumber(w)
	h = tonumber(h)
	l = tonumber(l)
	a, b, c = w * h, h * l, l * w
	p1 = p1 + 2 * a + 2 * b + 2 * c + math.min(a, b, c)
	a, b, c = w + h, h + l, l + w
	p2 = p2 + math.min(2 * a, 2 * b, 2 * c) + w * h * l
end

print("A: "..p1)
print("B: "..p2)
